tomviz/CentralWidget.cxx

```diff
@@ -325,23 +351,15 @@ void CentralWidget::setDataSource(DataSource* source)
     }
 
   // Calculate a histogram.
-  vtkNew<vtkTable> table;
+  vtkSmartPointer<vtkTable> table =
+    vtkSmartPointer<vtkTable>::New();
   this->HistogramCache[image] = table.Get();
+  vtkSmartPointer<vtkImageData> const imageSP = image;
 
-  if (!this->Worker)
-    {
-    this->Worker = new HistogramWorker(this);
-    connect(this->Worker, SIGNAL(finished()), SLOT(histogramReady()));
-    }
-  else if (this->Worker->isRunning())
-    {
-    // FIXME: Queue, abort, something.
-    qDebug() << "Worker already running, skipping this one.";
-    return;
-    }
-  this->Worker->input = image;
-  this->Worker->output = table.Get();
-  this->Worker->start();
+  QMetaObject::invokeMethod(this->HistogramGen,
```

Comment:
Here, I want the method to be called on the background worker thread rather than the main event loop thread.  invokeMethod schedules the method to be called on the event loop of the thread that owns the object.  I don't want to call the method directly on the GUI thread since histogram creation is expensive.